src/azul/service/responseobjects/storage_service.py

```diff
@@ -1,34 +1,38 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
 from logging import getLogger
-from typing import Optional
+from typing import Optional, List
 import boto3
 from azul import config
 
 logger = getLogger(__name__)
+AWS_S3_DEFAULT_MINIMUM_PART_SIZE = 5242880  # 5 MB; see https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
+MULTIPART_UPLOAD_MAX_WORKERS = 12
 
 
 class StorageService:
 
     def __init__(self):
-        self.__bucket_name = config.s3_bucket
-        self.__client = None  # the default client will be assigned later to allow patching.
+        self.bucket_name = config.s3_bucket
```

Comment:
See I don't understand why `bucket_name` is public while `_client` is protected. It could have disastrous consequences if someone modified `bucket_name` on an existing instance. Assuming I am correct, if we go to the extend of distinguishing between public and protected, we need to do so correctly, otherwise all that syntactic noise that the distinction introduces just makes things worse.

Reply:
Originally, I only made `_client` protected for lazy loading. However, if `self._client` was removed as the result of using `lru_cache` on `self.client` (property), should we still keep `bucket_name` public?